pandas/core/generic.py

```diff
@@ -5799,6 +5800,112 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    def convert_dtypes(
+        self: FrameOrSeries,
+        use_nullable_dtypes: bool_t = True,
+        convert_integer: bool_t = True,
+    ) -> FrameOrSeries:
+        """
+        Convert columns to best possible dtypes, optionally using dtypes supporting
+        ``pd.NA``.
+
+        For object-dtyped columns, use the inference rules as during normal
+        Series/DataFrame construction.  Then, if possible, convert to ``StringDtype``,
+        ``BooleanDtype`` or an appropriate integer extension type, otherwise leave as
+        ``object``.
+
+        If the dtype is integer, convert to an appropriate integer extension type.
+
+        If the dtype is numeric, and consists of all integers, convert to an
+        appropriate integer extension type.
+
+        .. versionadded:: 1.0.0
+
+        Parameters
+        ----------
+        use_nullable_dtypes : bool, default True
+            Whether conversion to types supporting ``pd.NA`` should be attempted.
+        convert_integer : bool, default True
+            If ``use_nullable_dtypes`` is True, Whether ``int`` types should be
```

Comment:
what is the case for convert_integer=Ture and use_nullable_dtypes=False and the converse? why do you have 2 options here which overlap? arguments need to be completely orthogonal.

This makes for a not-great user experience.
```(Ignored if ``use_nullable_dtypes`` is False```

Reply:
fixed

Reply:
@Dr-Irv how did you "fix" this?

Reply:
@jorisvandenbossche New signature is:
```python
   def convert_dtypes(
        self: FrameOrSeries,
        infer_objects: bool_t = True,
        convert_string: bool_t = True,
        convert_integer: bool_t = True,
        convert_boolean: bool_t = True,
    ) -> FrameOrSeries:
```